app/lib/actions/remote_execution/run_host_job.rb

```diff
@@ -121,6 +125,16 @@ def exit_status
         delegated_output[:exit_status]
       end
 
+      # Method that produces hash where key is output type (ie. stdout, stderr...) and value is output for given type
+      # So the returned hash look like {'stdout': 'normal output', 'stderr': 'error'}
+      def output_by_type
```

Comment:
@adamruzicka could you ack that this method make sense, I'm not sure here.

Reply:
It gets to the result in a bit roundabout way, but essentially yes.

Although I think that having
```json
{ "stdout": ["a", "c"], "stderr": ["b"] }
```

is less useful than

```
{
  "outputs": [
    {"type": "stdout", "value": "a"},
    {"type": "stderr", "value": "b"},
    {"type": "stdout", "value": "c"},
  ]
}
```
because you lose information about ordering

Reply:
It's little worse, with the current method implementation, I only get the last value for given type. So i this case, the result is: `{  "stdout": "c", "stderr": "b" }`. 

I didn't know that this can happen (seems logically right know) but still I've tried to display this values in the report. In worst case, in report that is excel table and value is in cell. Honestly, I didn't get any suitable solution for this. I have some ideas, but I don't think that add "position" number to result is good idea. Also showing one task on host for more times for given outputs is not good idea to me.  

Any opinions in there? @adamruzicka @ezr-ondrej @ares

Reply:
Keeping just last output from each type is a no go I'm afraid.

Can't we go with the format I proposed (keeping ordering and type) and then deal with the serialization in the template? If the source data allows us to easily transform it then we could quite easily:
1) Discard the type, join all the outputs into one string with newlines - keeps ordering, discards type
2) Join per-type outputs into strings - loses ordering, keeps type
3) Prefix each output with type specifier (`STDOUT | $message`), join them with newlines - keeps ordering and type, but the reported output as a raw string won't be exactly the same as what came out of the job

Reply:
AND we could do this dirrectly on `ForemanTasksCore::ContinuousOutput`.
We could do:
```ruby
# class ForemanTasksCore::ContinuousOutput
  def to_json
    sort!
    # format raw_outputs to json
  end
```

And have this method just as a proxy to that.